pkg/manager/member/pump_member_manager.go

```diff
@@ -181,11 +181,12 @@ func (m *pumpMemberManager) syncHeadlessService(tc *v1alpha1.TidbCluster) error
 
 func (m *pumpMemberManager) syncConfigMap(tc *v1alpha1.TidbCluster, set *appsv1.StatefulSet) (*corev1.ConfigMap, error) {
 
-	basePumpSpec, createPump := tc.BasePumpSpec()
-	if !createPump {
+	if tc.Spec.Pump == nil {
```

Comment:
This is already checked in https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pump_member_manager.go#L54.

Reply:
fixed